lib/generate_functions.py

```diff
@@ -225,7 +225,16 @@ def parse_header(header_name: str, output_file: str, ext_file: str, prefix: str,
         if not line.startswith(prefix):
             continue
 
-        line = line.split(";", 1)[0]
+        split_line = line.split(";", 1)
+
+        line = split_line[0]
+        inline_comment = ""
+        if len(split_line) > 1:
+            inline_comment = '/' + split_line[1].lstrip()
+            # sanity check
+            if inline_comment == "/":
+                inline_comment = ""
```

Comment:
I would much rather we check for empty comments before settings `inline_comment`, instead of setting and unsetting it. Maybe something like
```py
split_line[1] = split_line[1].lstrip()
if split_line.startswith("//"):
    inline_comment = split_line[1]
```

Reply:
Yes sorry, my implementation was hasty